Property 'applicationName' and 'context' of type 'string | undefined' is not assignable to 'string' index type 'string'

**node.js**
```
$ node -v
v22.11.0
```

**npm**
```
$ npm -v
10.9.0
```

**Packages**
```
    "@angular/animations": "^18.2.13",
    "@angular/cdk": "^18.2.14",
    "@angular/common": "^18.2.13",
    "@angular/compiler": "^18.2.13",
    "@angular/core": "^18.2.13",
    "@angular/platform-browser": "^18.2.13",
    "@angular/platform-browser-dynamic": "^18.2.13",
    "@angular/router": "^18.2.13",
    "@angular/service-worker": "^18.2.13",
    "@cds/core": "^6.14.0",
    "@clr/angular": "^17.4.0",
    "@clr/ui": "^17.4.0",
    "@porscheinformatik/clr-addons": "^18.1.8",
```

**Problem**
Immediately after integration I am getting the following error:
```
Property 'applicationName' and 'context' of type 'string | undefined' is not assignable to 'string' index type 'string'
```

**some.module.ts**
```
import { ClarityModule } from '@clr/angular';
import { ClrAddonsModule } from '@porscheinformatik/clr-addons';

@NgModule({
  imports: [
    ClarityModule,
    ClrAddonsModule,
  ],
})
export class SomeModule {}
```

**Error**
```
X [ERROR] TS2411: Property 'applicationName' of type 'string | undefined' is not assignable to 'string' index type 'string'. [plugin angular-compiler]

    node_modules/@porscheinformatik/clr-addons/history/history-model.interface.d.ts:16:8:
      16 │         applicationName?: string;
         ╵         ~~~~~~~~~~~~~~~


X [ERROR] TS2411: Property 'context' of type 'string | undefined' is not assignable to 'string' index type 'string'. [plugin angular-compiler]

    node_modules/@porscheinformatik/clr-addons/history/history-model.interface.d.ts:18:8:
      18 │         context?: string;
```

From:
```
    context: {
        applicationName?: string;
        tenantid: string;
        context?: string;
        [key: string]: string;
    };
```

To - Version 1:
```
    context: {
        applicationName?: string;
        tenantid: string;
        context?: string;
        [key: string]: string | undefined;
    };
```

To - Version 2:

```
    context: {
        applicationName: string;
        tenantid: string;
        context: string;
        [key: string]: string;
    };
```

--- a/src/clr-addons/history/history-model.interface.ts
+++ b/src/clr-addons/history/history-model.interface.ts
@@ -26,7 +26,7 @@ export interface ClrHistoryModel {
     applicationName?: string;
     tenantid: string;
     context?: string;
-    [key: string]: string;
+    [key: string]: string | undefined;
   };
 
   /**

